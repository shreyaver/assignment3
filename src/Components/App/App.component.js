import React, { Component } from 'react';
import Header from '../Header/Header.component';
import Books from '../Books/Books.component';
import './App.component.css';
import getRequest from '../../Helpers/Request/request';
import { ALL_BOOKS_URL } from '../../config';

class App extends Component {
  state = { books: {} };

  componentDidMount() {
    getRequest(ALL_BOOKS_URL).then((books) => {
      this.setState({ books: books.data });
    });
  }

  render() {
    const { books } = this.state;
    return (
      <div className="app">
        <Header />
        <Books books={books} />
      </div>
    );
  }
}

export default App;
