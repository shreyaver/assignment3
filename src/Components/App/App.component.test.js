import React from 'react';
import renderer from 'react-test-renderer';
import App from './App.component';

describe('App', () => {
  it('renders without crashing', () => {
    const landingPage = renderer.create(<App />).toJSON();
    expect(landingPage).toMatchSnapshot();
  });
});
