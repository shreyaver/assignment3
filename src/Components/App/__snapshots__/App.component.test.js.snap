// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div
  className="app"
>
  <header>
    <div
      className="headerContent"
    >
      <div
        className="circle"
      >
        BS
      </div>
    </div>
  </header>
  <div
    className="books"
  >
    <div
      className="booksContent"
    />
  </div>
</div>
`;
